validation.js

```diff
@@ -0,0 +1,148 @@
+console.log('validation.js loaded check');
+document.addEventListener('DOMContentLoaded', function () {
+    const emailLogin = document.getElementById('email-login');
+    const passwordLogin = document.getElementById('password-login');
+    const loginButton = document.getElementById('login-btn');
+
+    const emailSignup = document.getElementById('email-signup');
+    const nicknameSignup = document.getElementById('nickname');
+    const passwordSignup = document.getElementById('password-signup');
+    const passwordConfirmationSignup = document.getElementById('password-confirmation');
+    const signupButton = document.getElementById('signup-btn');
+
+    function validateEmail(email) {
+        const regex = /^[^\s@]+@[^\s@]+.[^\s@]+$/;
+
+        return regex.test(email);
+    }
+
+    function validatePassword(password) {
+        return password.length >= 8;
+    }
+
+    function toggleButtonState() {
+        const isValidLogin =
+            emailLogin?.value &&
+            passwordLogin?.value &&
+            validateEmail(emailLogin.value) &&
+            validatePassword(passwordLogin.value);
+        if (loginButton) loginButton.disabled = !isValidLogin;
+
+        const isValidSignup =
+            emailSignup?.value &&
+            nicknameSignup?.value &&
+            passwordSignup?.value &&
+            passwordConfirmationSignup?.value &&
+            validateEmail(emailSignup.value) &&
+            validatePassword(passwordSignup.value) &&
+            passwordSignup.value === passwordConfirmationSignup.value;
+        if (signupButton) signupButton.disabled = !isValidSignup;
+    }
```

Comment:
💊 제안
이런경우 가독성을 위해 따로 함수로 빼면 좋을 것 같아요.

```suggestion
        // 로그인, 회원가입 로직 분리시 코드
        signupButton.disabled = !checkFormValid();
    }
```